pyuvsim/simsetup.py

```diff
@@ -836,7 +836,77 @@ def _sky_select_calc_rise_set(sky, source_params, telescope_lat_deg=None):
     return sky
 
 
-def initialize_catalog_from_params(obs_params, input_uv=None, return_recarray=True):
+def _read_pyradiosky_file(filename, source_params):
+    """
+    Read in a pyradiosky catalog file.
+
+    Parameters
+    ----------
+    filename : str
+        File to read in.
+    source_params : dict
+        Dictionary specifying options to pyradiosky read methods.
+
+    Returns:
+    --------
+    :class:`pyradiosky.SkyModel`
+        SkyModel object based on filename
+
+    """
+    sky = pyradiosky.SkyModel()
+
+    if filename.endswith("txt"):
+        sky.read_text_catalog(filename)
+    elif filename.endswith('vot'):
+        if 'gleam' in filename:
+            if "spectral_type" in source_params:
+                spectral_type = source_params["spectral_type"]
+            else:
+                warnings.warn("No spectral_type specified for GLEAM, using 'flat'.")
+                spectral_type = "flat"
+            sky.read_gleam_catalog(
+                filename, spectral_type=spectral_type
+            )
```

Comment:
this seems somewhat dangerous... what if the file is GLEAM but named something like "from_natasha.vot"? I guess the user can always just read it themselves if they have a weird special case. Maybe we should at least use `'gleam' in filename.lower()`

Reply:
I added an optional `filetype` parameter to this method (and the one that calls it) to allow users to specify the filetype rather than relying on this parsing. It falls back on this parsing if the filetype is not specified.

I also added the `.lower()` for checking for gleam.